sympy/matrices/matrices.py

```diff
@@ -877,6 +877,136 @@ def rank(self, iszerofunc=_iszero, simplify=False):
         echelon_form, pivots, swaps = mat._eval_echelon_form(iszerofunc=iszerofunc, simpfunc=simpfunc)
         return len(pivots)
 
+    def rank_decomposition(self,
+        iszerofunc=_iszero, simplify=False, format='original'):
+        r"""Returns the rank-factorized form of `A`, as `C` and `F`.
+
+        Parameters
+        ==========
+
+        iszerofunc : Function, optional
+            A function used for detecting whether an element can
+            act as a pivot.  ``lambda x: x.is_zero`` is used by default.
+
+        simplify : Bool or Function, optional
+            A function used to simplify elements when looking for a
+            pivot. By default SymPy's ``simplify`` is used.
+
+        format : String, optional
```

Comment:
I don't know how I feel about this parameter name. Also, is this parameter needed? It sounded like the application you outlined would always use `format="reduced"`

Reply:
From my read of this, I don't think it's necessary to keep. It adds nothing to the utility of the result. I would recommend dropping the keyword and always use the reduced form.

I also made some edits which you can feel free to keep, modify or revert. :-)

Reply:
If we may use reduced formatting, it may break some formatting consistency with `rref` though,
but I have made it to reduced, anyway.